File: app/src/main/java/com/bykea/pk/partner/ui/withdraw/DialogWithdrawConfirmation.java
package com.bykea.pk.partner.ui.withdraw;

import android.content.Context;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import androidx.annotation.NonNull;

import com.bykea.pk.partner.R;
import com.google.android.material.bottomsheet.BottomSheetDialog;

public class DialogWithdrawConfirmation extends BottomSheetDialog {
    private WithdrawalViewModel withdrawalViewModel;

    public DialogWithdrawConfirmation(@NonNull Context context) {
        super(context);
    }

    public DialogWithdrawConfirmation(@NonNull Context context, int theme) {
        super(context, theme);
    }


    protected DialogWithdrawConfirmation(@NonNull Context context, boolean cancelable, OnCancelListener cancelListener) {
        super(context, cancelable, cancelListener);
    }

    public static DialogWithdrawConfirmation newInstance(WithdrawalActivity withdrawalActivity,
                                                         WithdrawalViewModel withdrawalViewModel) {
        DialogWithdrawConfirmation dialogWithDrawConfirmation = new DialogWithdrawConfirmation(withdrawalActivity);
        dialogWithDrawConfirmation.setViewModel(withdrawalViewModel);
        return dialogWithDrawConfirmation;
    }

    private void setViewModel(WithdrawalViewModel withdrawalViewModel) {
        this.withdrawalViewModel = withdrawalViewModel;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.dialog_withdraw_confirmation);

        View dismissTextView = findViewById(R.id.dismiss_withdraw_txtview);
        View confirmTextView = findViewById(R.id.confirm_withdraw_txtview);

        TextView totalTextView = findViewById(R.id.confirmation_total_value);
        TextView feesTextView = findViewById(R.id.confirmation_fees_value);
        TextView amountTextView = findViewById(R.id.confirmation_payment_value);

        int amount = withdrawalViewModel.getBalanceToWithdraw().getValue();
        long fees = Math.round(withdrawalViewModel.getSelectedPaymentMethod().getFees());

        amountTextView.setText(String.valueOf(amount));
        feesTextView.setText(String.format("-%s", fees));
        totalTextView.setText(String.valueOf(amount + fees));

        dismissTextView.setOnClickListener(v -> withdrawalViewModel.getShowConfirmationDialog().setValue(false));
        confirmTextView.setOnClickListener(v -> {
            withdrawalViewModel.getShowConfirmationDialog().setValue(false);
            withdrawalViewModel.confirmWithdraw();
        });
    }
}

File: app/src/main/java/com/bykea/pk/partner/ui/withdraw/DialogWithdrawConfirmation.kt
package com.bykea.pk.partner.ui.withdraw

import android.app.Activity
import android.view.View
import android.widget.TextView
import androidx.databinding.DataBindingUtil
import com.bykea.pk.partner.R
import com.bykea.pk.partner.databinding.DialogWithdrawConfirmationBinding
import com.google.android.material.bottomsheet.BottomSheetDialog

/**
 * This class will responsible to show withdrawal summary on a dialog
 *
 * @author Arsal Imam
 */
open class DialogWithdrawConfirmation : BottomSheetDialog {

    /**
     * view binder all ui components present in xml
     */
    private var viewBinder: DialogWithdrawConfirmationBinding? = null

    /**
     * viewModel of this view
     */
    private var withdrawalViewModel: WithdrawalViewModel? = null

    /**
     * contructor to create new instance of this class
     */
    constructor(activity: Activity) : super(activity) {
        initUi(activity = activity)
    }

    /**
     * this method will initialize the bottom sheet UI components
     */
    private fun initUi(activity: Activity) {

        viewBinder = DataBindingUtil.inflate(activity.layoutInflater, R.layout.dialog_withdraw_confirmation,
                        null, false)

        ownerActivity = activity
        window.setBackgroundDrawableResource(android.R.color.transparent)
        setContentView(viewBinder?.root)

        viewBinder?.dismissWithdrawTxtview?.setOnClickListener { withdrawalViewModel?.showConfirmationDialog(false) }
        viewBinder?.confirmWithdrawTxtview?.setOnClickListener {
            withdrawalViewModel?.showConfirmationDialog(false)
            withdrawalViewModel?.confirmWithdraw()
        }
    }

    /**
     * Setter of viewModel object
     */
    private fun setViewModel(withdrawalViewModel: WithdrawalViewModel?) {
        this.withdrawalViewModel = withdrawalViewModel
    }

    /**
     * open dialog and update the content of the dialog
     */
    fun showDialog() {
        show()
        updateContent()
    }

    /**
     * update dialog everytime on popup opens...
     */
    private fun updateContent() {
        if (viewBinder == null) return

        val amount = withdrawalViewModel?.balanceInt?.value!!
        val fees = Math.round(withdrawalViewModel?.selectedPaymentMethod?.fees!!)

        viewBinder!!.confirmationPaymentValue?.text =
                String.format(ownerActivity.getString(R.string.formatted_price), amount)

        viewBinder!!.confirmationFeesValue?.text =
                String.format(ownerActivity.getString(R.string.specifier_string), fees)

        viewBinder!!.confirmationTotalValue?.text =
                String.format(ownerActivity.getString(R.string.rs_price), (amount - fees))
    }

    companion object {

        /**
         * This will return a new instance of this class
         *
         * @param withdrawalActivity context to launch new instance
         * @param withdrawalViewModel viewModel
         */
        fun newInstance(withdrawalActivity: WithdrawalActivity,
                        withdrawalViewModel: WithdrawalViewModel?): DialogWithdrawConfirmation {
            val dialogWithDrawConfirmation = DialogWithdrawConfirmation(withdrawalActivity)
            dialogWithDrawConfirmation.setViewModel(withdrawalViewModel)
            return dialogWithDrawConfirmation
        }
    }
}